torchci/lib/bot/greenBot.ts

```diff
@@ -0,0 +1,231 @@
+import { Probot, ProbotOctokit } from "probot";
+import { reactOnComment, addLabels } from "./botUtils";
+import getRocksetClient from "lib/rockset";
+import rocksetVersions from "rockset/prodVersions.json";
+
+const openOnGreen = new RegExp("^s*@pytorchbot\\s+?mergeOnGreen.*");
+export const LAND_PENDING = "land-pending";
+const LAND_ATTEMPT = "landed-attempt";
+const LAND_FAILED = "land-failed";
+
+type OctokitData = {
+  octokit: InstanceType<typeof ProbotOctokit>;
+  owner: string;
+  repo: string;
+  issue_number: number;
+  comment: string;
+};
+
+export default function greenBot(app: Probot): void {
+  app.on("issue_comment.created", async (ctx) => {
+    const commentBody = ctx.payload.comment.body;
+    if (commentBody.match(openOnGreen)) {
+      if (!ctx.payload.issue.pull_request) {
+        // Issue, not pull request.
+        await reactOnComment(ctx, "confused");
+      } else {
+        await reactOnComment(ctx, "+1");
+        await addLabels(ctx, [LAND_PENDING]);
+      }
+    }
+  });
+
+  app.on(["pull_request.labeled"], async (ctx) => {
+    if (ctx.payload.label.name === LAND_PENDING) {
+      // Check pull request's head ref and check runs to see if it has all green and merge if it's good
```

Comment:
Wait, why do we do this instead of allowing trymerge.py to check against the required status checks?

I agree this parsing behavior may be necessary for future "on green", but the current functionality checks against all CI instead of just required status check CI, no?

Reply:
It checks that all jobs have succeed before merging, which is what we wanted right? mergeOnGreen to me meant that all signals have to be green. If we want to have it only be the required ones, we can change it to that or add a flag.

Reply:
We chatted about this and if we move all the validation logic into the try-merge, then we potentially have to run that workflow after every workflow_job right? I think just doing some lightweight checks like the statuses here before kicking off that workflow is a bit better. I think we still want to leave the more complicated merge_rules logic in that script.

For now we were thinking of just having a list of some of the required checks here and eventually making everything required.

Reply:
Could be like:

1. Comment does workflow dispatch for trymerge. 

2. Trymerge just spins and queries the aggregate workflow jobs endpoint every minute or so to check that the required jobs are successful.

So we wouldn't need to trigger after each workflow job

Reply:
oh huh is having trymerge spin for a really long time better than retriggering it for every workflow job finish? For now trymerge wouldn't be spinning a long time, but when we make more things required, or for the +all command, trymerge could be spinning for 3 hours, no?

Reply:
Yes the merge workflow could take 3+h. But…we are proposing to make the act of merging potentially take 3+h, so I don't really see what's wrong with that lol. If anything, I think it makes things more clear? (If you didn't know anything about the system, you probably would expect trymerge to try to merge and fail if you didn't merge correctly).